### ISSUE TITLE
CICE C-grid runs using 0.25deg

### ISSUE THREAD

**Issue by anton-seaice** [ACCESS-NRI/access-om3-configs#409]
Following recent discussions , I ran the 025deg ryf with 

Diabatic_first=True
Dt_therm = 5400s (1.5 hours)

and otherwise the default configuration (the om2 grid / bathymetry / initial conditions / cice b-grid) 

It ran until the 18th March in the second year

And crashed with too many truncations :

```
FATAL from PE   379: write_energy : Ocean velocity has been truncated too many times.

Image              PC                Routine            Line        Source             
access-om3-MOM6-C  0000000002E3BB14  mpp_mod_mp_mpp_er          72  mpp_util_mpi.inc
access-om3-MOM6-C  0000000001E4F82F  mom_error_handler         154  MOM_error_handler.F90
access-om3-MOM6-C  0000000001D4F9DF  mom_sum_output_mp         904  MOM_sum_output.F90
access-om3-MOM6-C  0000000001C0E444  mom_mp_step_mom_         1016  MOM.F90
access-om3-MOM6-C  0000000001BAE725  mom_ocean_model_n         624  mom_ocean_model_nuopc.F90
access-om3-MOM6-C  0000000001B780C2  mom_cap_mod_mp_mo        1883  mom_cap.F90
access-om3-MOM6-C  0000000000AE1C88  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000000AE1BEF  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000000AE07A2  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000000C47E82  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000003B67E20  Unknown               Unknown  Unknown
access-om3-MOM6-C  00000000009F0FB2  Unknown               Unknown  Unknown
access-om3-MOM6-C  00000000009F4B0F  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000000B4B39A  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000000B37EE5  Unknown               Unknown  Unknown
access-om3-MOM6-C  00000000009F23EA  Unknown               Unknown  Unknown
access-om3-MOM6-C  000000000050CACA  Unknown               Unknown  Unknown
access-om3-MOM6-C  0000000000759361  Unknown               Unknown  Unknown
access-om3-MOM6-C  00000000009666A7  Unknown               Unknown  Unknown
...
```

Is this consistent with your work @minghangli-uni - should I increase the number of allowable truncations and see if they stabilise?

Config is here: https://github.com/ACCESS-NRI/access-om3-configs/compare/34e2136...51fbffa1d152d7773093c9d26924fd0d08fa5880
Output is in:

`/g/data/tm70/as2285/payu/dev-025deg_jra55do_ryf-bgrid/work`

**Comment by anton-seaice** [comment]
I ran three variants based on [0.5-alpha](https://github.com/ACCESS-NRI/access-om3-configs/releases/tag/dev-MC_25km_jra55_ryf_0.5) and using the 49 year of spun up ocean (/g/data/tm70/ml0072/COMMON/git_repos/test-Reichl-2025-04-continue/archive/restart019_output) :

- c-grid
- c-grid and ndte=240
- c-grid and revp=True

They all crashed with  this error at different times and locations

```
(abort_ice)ABORTED: 
 (abort_ice) error = (diagnostic_abort)ERROR: negative area (open water)
 ERROR: (abort_ice)(diagnostic_abort)ERROR: negative area (open water)
```

I haven't investigated further

**Comment by aekiss** [comment]
Looks like there's ~100% coverage in the thickest category. What's the thickness in that cell?

**Comment by anton-seaice** [comment]
This looks a lot like the same issue with the ridging parameterisation as https://github.com/CICE-Consortium/CICE/issues/992

<img width="558" height="453" alt="Image" src="https://github.com/user-attachments/assets/c8909249-d508-44ad-bf35-7bc93de8a704" />

I'll re-run with some extra diagnostics to confirm. I was hoping moving to the 900s timestep would have fixed the issue. It may mean we need to start thinking about basal stress, increasing ndtd, or changing the ridging parameterisation. (JF [notes](https://github.com/CICE-Consortium/CICE/issues/992#issuecomment-2506215655) about ECCC models : 

> With kstrength=1, we get unrealistic thicknesses in tidally active regions (it might be the same problem that is described here...). Thicknesses are more realistic with kstrength=0. We use Pstar=2.25e4. 

I don't really understand how our tidal parameterisation works, would that impact the ocean-> sea ice stress in some way ? Possibly these fields:

```
             strtltxU=  -0.806291062699686     
             strtltyU=   0.837399882196317     
```

**Comment by anton-seaice** [comment]
I think i might have made some progress here.

I noticed the crash location is a high volume river in the Arctic, right when seasonal runoff is increasing. We also get unrealistic thickness there (similar to the quote from JF above). Turning off river runoff runoff passes the crash time without issue.

**Comment by aekiss** [comment]
Hm, I wonder whether horizontal runoff redistribution will help? https://github.com/ACCESS-NRI/access-om3-configs/issues/396

**Comment by anton-seaice** [comment]
> Turning off river runoff runoff passes the crash time without issue.

I also re-ran with rivers on and with CICE using internally calculated SSH stress (instead of coupled) and it doesn't crash. Seems to confirm that it's related to the high volume of runoff (and that CM3 might not have the issue due to more spreading @kieranricardo )

**Comment by anton-seaice** [comment]
It seems like the runoff distribution was incidental, this runs now after fixing the grid angle (#1289 ) and without the runoff distribution 

This ran for 10 years without crashing, the results re a bit hard to interpret to separate the impact of fixing the grid angle from cice c-grid.

Orange = old, blue = new:

<img width="988" height="468" alt="Image" src="https://github.com/user-attachments/assets/1751760d-9f94-428f-b9d9-0d78ffedd064" />

The old (test4) results were thought were probably too thick, so this makes sense.

This is the diff in arctic volumes (c-grid, with correct angles minus test4)

<img width="506" height="409" alt="Image" src="https://github.com/user-attachments/assets/527d4a9c-4143-4e6c-9fbe-649bf179f4bc" />

The maximum thickness decreases:

<img width="997" height="258" alt="Image" src="https://github.com/user-attachments/assets/bf96dee7-b435-4098-b8d3-87ee450ab9d2" />

So this seems like enough to progress with c-grid, we can discover the other issues as time progressed.

Ill try removing the b-grid polar mask now, and see what happens then

**Comment by anton-seaice** [comment]
There's a few bays where ice gets thick without the b-grid land mask in the polar region:

987  1090  63.92143249511719  0.0

<img width="873" height="374" alt="Image" src="https://github.com/user-attachments/assets/8500bfbd-3cc9-4ea4-8280-cf250db9d151" />

351  125  103.62161254882812  0.0

<img width="873" height="371" alt="Image" src="https://github.com/user-attachments/assets/9d40843b-af00-488a-8f86-5e18c8e36c3c" />

846  171  317.72381591796875  0.0

<img width="873" height="371" alt="Image" src="https://github.com/user-attachments/assets/5974c686-7ea7-4d2e-a01d-a45fd69baa5a" />

I guess its safe to edit those out

**Comment by aekiss** [comment]
> I guess its safe to edit those out

Yep! Maybe remove the whole 1-cell-wide embayments in the first 2 cases.

How long was this run? I'm wondering whether 7m is a lot or not.

**Comment by anton-seaice** [comment]
Only ~7 years. These are points with mean thickness > 6m over years 2-7. Ill recheck them when it gets to 10 years.

(There's several more points, if only looking at the end of the run. This is probably conservative, and would need a full cycle to really know where the ice has got stuck)

**Comment by anton-seaice** [comment]
Thick ice points look basically the same after 10 years, there's one more point which jumps out:

1269  172  143.9140167236328  0.0

<img width="873" height="376" alt="Image" src="https://github.com/user-attachments/assets/87803f02-5312-4da6-a96e-fb4b503f67e6" />

For whatever reason, some single cell wide channels work fine:

<img width="873" height="374" alt="Image" src="https://github.com/user-attachments/assets/bb14a871-5f9d-4e14-aa1f-e592fa0c39c4" />

Removing the b-grid polar mask appears to give results basically the same as with the polar b-grid mask, so ill plan on putting b-grid non advective points back in.

There's some example output in /g/data/tm70/as2285/payu/cgrid-aug26/access-om3-configs-c-grid-mask

**Comment by anton-seaice** [comment]
Implemented in

| PR | Branch | Status |
|---|---|---|
| #1596 | `dev-MC_25km_jra_iaf` | ✅ merged |
| #1598 | `dev-MC_25km_jra_iaf+wombatlite` | ✅ merged |
| #1599 | `dev-MC_25km_jra_ryf` | ✅ merged |
| #1600 | `dev-MC_25km_jra_ryf+wombatlite` | ✅ merged |

The new input files / bathymetry included in that change includes the points where sea ice advections isn't possible on the B-grid. [Test 5](https://github.com/ACCESS-NRI/access-om3-configs/issues/1534) should clarify how well that works.

**Comment by chrisb13** [comment]
Hurrah @anton-seaice! 

I know we discussed yesterday rolling this out across the 25k's first but do we have a mechanism by which we'll remember to eventually update the other configs?

**Comment by anton-seaice** [comment]
https://github.com/ACCESS-NRI/access-om3-configs/issues/245 is still open, I think that's enough for now
